drivers/media/platform/raspberrypi/pisp_be/pisp_be.c

```diff
@@ -40,32 +34,36 @@ MODULE_LICENSE("GPL v2");
 #define PISPBE_NAME "pispbe"
 
 /* Some ISP-BE registers */
-#define PISP_BE_VERSION_OFFSET (0x0)
-#define PISP_BE_CONTROL_OFFSET (0x4)
-#define PISP_BE_TILE_ADDR_LO_OFFSET (0x8)
-#define PISP_BE_TILE_ADDR_HI_OFFSET (0xc)
-#define PISP_BE_STATUS_OFFSET (0x10)
-#define PISP_BE_BATCH_STATUS_OFFSET (0x14)
-#define PISP_BE_INTERRUPT_EN_OFFSET (0x18)
-#define PISP_BE_INTERRUPT_STATUS_OFFSET (0x1c)
-#define PISP_BE_AXI_OFFSET (0x20)
-#define PISP_BE_CONFIG_BASE_OFFSET (0x40)
-#define PISP_BE_IO_INPUT_ADDR0_LO_OFFSET (PISP_BE_CONFIG_BASE_OFFSET)
-#define PISP_BE_GLOBAL_BAYER_ENABLE_OFFSET (PISP_BE_CONFIG_BASE_OFFSET + 0x70)
-#define PISP_BE_GLOBAL_RGB_ENABLE_OFFSET (PISP_BE_CONFIG_BASE_OFFSET + 0x74)
-#define N_HW_ADDRESSES 14
-#define N_HW_ENABLES 2
-
-#define PISP_BE_VERSION_2712C1 0x02252700
-#define PISP_BE_VERSION_MINOR_BITS 0xF
+#define PISP_BE_VERSION_REG			0x0
+#define PISP_BE_CONTROL_REG			0x4
+#define PISP_BE_CONTROL_COPY_CONFIG		BIT(1)
+#define PISP_BE_CONTROL_QUEUE_JOB		BIT(0)
+#define PISP_BE_CONTROL_NUM_TILES(n)		((n) << 16)
+#define PISP_BE_TILE_ADDR_LO_REG		0x8
+#define PISP_BE_TILE_ADDR_HI_REG		0xc
+#define PISP_BE_STATUS_REG			0x10
+#define PISP_BE_STATUS_QUEUED			BIT(0)
+#define PISP_BE_BATCH_STATUS_REG		0x14
+#define PISP_BE_INTERRUPT_EN_REG		0x18
+#define PISP_BE_INTERRUPT_STATUS_REG		0x1c
+#define PISP_BE_AXI_REG				0x20
+#define PISP_BE_CONFIG_BASE_REG			0x40
+#define PISP_BE_IO_INPUT_ADDR0(n)		(0x40 + 8 * (n))
+#define PISP_BE_GLOBAL_BAYER_ENABLE		0xb0
+#define PISP_BE_GLOBAL_RGB_ENABLE		0xb4
+#define N_HW_ADDRESSES				14
+#define N_HW_ENABLES				2
+
+#define PISP_BE_VERSION_2712C1			0x02252700
```

Comment:
The reference to C1 is strictly correct but confusing -- D0 differs only in the Minor bits, which we don't check.
I wonder if we should omit the `C1` suffix?